crates/anstyle-roff/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+//! roff output.
+
+mod roff_styles;
+mod style_stream;
+use roff::{bold, italic, Roff};
+
+#[derive(Debug, Default, Clone)]
+pub(crate) struct RoffStyle {
+    fg: Option<roff_styles::Color>,
+    bg: Option<roff_styles::Color>,
+    effects: anstyle::Effects,
+    text: Option<String>,
+}
+
+impl RoffStyle {
+    pub fn new() -> Self {
+        Self {
+            fg: None,
+            bg: None,
+            effects: anstyle::Effects::default(),
+            text: None,
+        }
+    }
+
+    /// Add text to be styled
+    pub fn text(&mut self, txt: String) -> &mut Self {
+        self.text = Some(txt);
+        self
+    }
+
+    fn fg(&mut self, fg: Option<roff_styles::Color>) -> &mut Self {
+        self.fg = fg;
+        self
+    }
+    fn bg(&mut self, bg: Option<roff_styles::Color>) -> &mut Self {
+        self.bg = bg;
+        self
+    }
+
+    fn effects(&mut self, effects: anstyle::Effects) -> &mut Self {
+        self.effects = effects;
+        self
+    }
+
+    /// Get the style out as a Roff type
+    pub(crate) fn as_roff(&self) -> Roff {
+        let mut doc = Roff::new();
+        doc.extend([set_color((&self.fg, &self.bg)), self.set_effects()]);
+        doc
+    }
+
+    fn set_effects(&self) -> Roff {
+        // Roff (the crate) only supports these inline commands
+        // Bold
+        // Italic
+        // Roman (plain text)
+        // If we want more support, or even support combined formats, we will need
+        // to push improvements to roff upstream or implement a more thorough roff crate
+        // perhaps by spinning off some of this code
+        let mut doc = Roff::new();
+        if let Some(ref txt) = self.text {
+            if self.effects.contains(anstyle::Effects::BOLD) {
+                doc.text(vec![bold(txt)]);
+                return doc;
+            }
+            if self.effects.contains(anstyle::Effects::ITALIC) {
+                doc.text(vec![italic(txt)]);
+                return doc;
+            }
+            if self.effects.is_plain() {
+                doc.text(vec![roff::roman(txt)]);
+                return doc;
+            }
+        }
+        doc
+    }
+}
+
+/// Generate A RoffStyle from Style
+///
+/// ```rust
+/// use anstyle::{Color, RgbColor};
+///
+/// let text = "\u{1b}[44;31mtest\u{1b}[0m";
+///
+/// let roff_doc = anstyle_roff::to_roff(text);
+/// let expected = r#".gcolor red
+/// .fcolor blue
+/// test
+/// "#;
+///
+/// assert_eq!(roff_doc.to_roff(), expected);
+/// ```
+pub fn to_roff(styled_text: &str) -> Roff {
+    let stream = style_stream::StyledStream::new(styled_text);
+
+    let mut roff_docs = vec![];
+    for style_str in stream {
+        let style = style_str.style;
+        let fg = ansi_color_to_roff(style.get_fg_color());
+        let bg = ansi_color_to_roff(style.get_bg_color());
+        let effect = style.get_effects();
+        let mut roff_style = RoffStyle::new();
```

Comment:
Do we still need `RoffStyle`?  Seems like it could be a function.

Reply:
Done